### PR TITLE
Move log registry to wagtailcore and validate log actions

### PR DESCRIPTION
**Pull request by kaedroho**
This PR moves the log registry into wagtailcore and adds validation to make sure log actions are registered at the point of inserting new logs. I've also renamed the registry so that we can have multiple registries for different object types.

This PR:

- [x] Move log registry to core
- [x] Make log registry specific to pages (so pages can have a different set of actions to images, for example)
- [x] Validate log action on save

Future PRs:

- Polymorphic log actions (where a `wagtail.copy` acts as a specific version of `wagtail.create`)
- Grouping logs (Logs that were generated at the same time are linked in the database and displayed as one)
- Auto-assign user if log generated in Wagtail admin (in case this is forgotten, or not possible due to the way the view is coded)

### CODE REVIEW

File: wagtail/core/tests/test_audit_log.py [hunk 1]
```diff
@@ -1,280 +1,265 @@
...
-        self.assertEqual(PageLogEntry.objects.get_for_user(self.user).count(), 1)  # the create from setUp
+        with self.register_hook('register_log_actions', test_hook):
+            log_actions = LogActionRegistry('register_log_actions')
+            self.assertIn('test.custom_action', log_actions.get_actions())
+            self.assertEqual(log_actions.format_message(log_entry), "Tested!")
+            self.assertEqual(log_actions.get_action_label('test.custom_action'), 'Custom action')
 
 
-class TestAuditLog(TestCase):
+class TestAuditLogAdmin(TestCase, WagtailTestUtils):
```
Comment (kaedroho) [inline wagtail/core/tests/test_audit_log.py:61]: Oops

File: wagtail/core/wagtail_hooks.py [hunk 2]
```diff
@@ -181,9 +181,9 @@ def rename_message(data):
         except KeyError:
             return _('Renamed')
 
-    actions.register_action('wagtail.rename', _('Rename'), rename_message)
-    actions.register_action('wagtail.revert', _('Revert'), revert_message)
-    actions.register_action('wagtail.copy', _('Copy'), copy_message)
+    actions.register_action('wagtail.rename', _('Rename'), rename_message, parent='wagtail.edit')
```
Comment (kaedroho) [inline wagtail/core/wagtail_hooks.py:184]: Should be separate.

File: wagtail/core/tests/test_audit_log.py [hunk 3]
```diff
@@ -24,19 +24,19 @@ def setUp(self):
     def test_register_log_actions_hook(self):
         # testapp/wagtail_hooks.py defines a 'blockquote' rich text feature with a hallo.js
```
Comment (jacobtoppm) [inline wagtail/core/tests/test_audit_log.py:26]: I don't think this comment applies here, unless I'm missing some rich text integration...

File: wagtail/core/tests/test_audit_log.py [hunk 4]
```diff
@@ -1,280 +1,265 @@
-from datetime import datetime, timedelta
+from datetime import timedelta
```
Comment (gasman) [inline wagtail/core/tests/test_audit_log.py:1]: It appears that you've moved wagtail/admin/tests/test_audit_log.py here, and overwritten the existing wagtail/core/tests/test_audit_log.py.

File: wagtail/core/tests/test_audit_log.py (hunk 3 above)
Comment (gasman) [reply wagtail/core/tests/test_audit_log.py:26]: This was already in the old code in wagtail/admin/tests/test_audit_log.py.

**Review by gasman — CHANGES_REQUESTED** [review]

**Comment by kaedroho** [comment]
Sorry for the mistake, this is ready for another review now!

**Review by gasman — APPROVED** [review]

File: wagtail/core/wagtail_hooks.py [hunk 5]
```diff
@@ -92,3 +93,193 @@ def describe_collection_children(collection):
...
+
+    def convert_alias_message(data):
+        try:
+            return _("Converted the alias '%(title)s' into a regular page") % {
+                'title': data['page']['title'],
+            }
+        except KeyError:
+            return _("Converted an alias into a regular page")
+
+    def move_message(data):
```
Comment (zerolab) [inline wagtail/core/wagtail_hooks.py:145]: note: this PR doesn't have the changes from #6762

**Comment by gasman** [comment]
Merged in 7423d72efe58d38e9a287999940b9b302cac58e4 + parents

**Pull request closed without merging**